ox3apiclient/__init__.py

```diff
@@ -348,13 +380,17 @@ def put(self, url, data=None):
         response.raise_for_status()
         return self._response_value(response)
 
-    def post(self, url, data=None):
+    def post(self, url, data):
         """Issue a POST request to url (either a full URL or API
         shorthand) with the data.
 
         """
         if self.api_path in JSON_PATHS:
-            response = self._session.post(self._resolve_url(url), data=json.dumps(data), timeout=self.timeout)
+            url = self._resolve_url(url)
+            post_url = url[1] + url[2] + url[3]
```

Comment:
it seems like a *very* bad idea to modify the public client in a way where a parameter called `url` does not accept an URL.